### PR TITLE
Shutdown implementation which hooks into /world/Del

### PR DESCRIPTION
**Pull request by ZephyrTFA**
Add shutdown handling for rust-g

Move log closing to said proc.
Implement job detaching for said proc.

### CODE REVIEW

File: dmsrc/log.dm [hunk 1]
```diff
@@ -1,2 +0,0 @@
-#define rustg_log_write(fname, text, format) RUSTG_CALL(RUST_G, "log_write")(fname, text, format)
```
Comment (ZeWaka) [inline dmsrc/log.dm:1]: no

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by AffectedArc07** [comment]
Wonder if this would fix our need to have a patch on the HTTP client that has a designated create and shutdown method.

File: dmsrc/world.dm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/world/Del(...)
+	RUSTG_CALL(RUST_G, "shutdown_rustg")
```
Comment (ZeWaka) [inline dmsrc/world.dm:2]: This'll end up just copying itself  in the rust_g.dm file right? It needs documentation on what it's doing and why it's doing it then.
Some codebases are particular about ordering. See: you don't want this before the `logs << "shutting down"` logging entry.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
where do they go when they get detatched.

**Comment by ZephyrTFA** [comment]
> where do they go when they get detatched.

No where, we just drop our reference to them which means that they become free standing threads. There's no easy way to hook cancelation logic in and leaving them orphaned won't cause any problems

**Comment by willox** [comment]
"Detaching" the threads would mean they may still exist after the module is unloaded which will result in a crash. It doesn't do anything.

**Pull request closed without merging**